frontend/src/views/WheresMyGene/components/HeatMap/components/Chart/index.tsx

```diff
@@ -207,8 +207,10 @@ export default memo(function Chart({
         };
 
         setAllChartProps((allChartProps) => {
-          allChartProps[tissue] = result;
-          return allChartProps;
+          const newAllChartProps = { ...allChartProps };
```

Comment:
@seve I changed this to copy a new object before mutation, just so that we don't mutate the original object! 

I also tested locally to make sure it does work lol Thanks!